app/workers/plos_details.rb

```diff
@@ -20,15 +20,33 @@ def perform(plos_paper)
                  "api_key=#{ENV.fetch('PLOS_API_KEY')}"
 
     detail_resp = Net::HTTP.get_response(URI.parse(detail_url))
+    # did we get a proper response?
+    unless detail_resp.is_a? Net::HTTPSuccess
+      # don't bother parsing broken or empty output
+      sleep 6
```

Comment:
That looks great and would be useful for all API-jobs! How about we put that into a second PR?